### PR TITLE
charlieplex fix interrupt and documentation for pull down config

### PR DESCRIPTION
**Pull request by tobiasarndt**
This PR has a fix for the issue with interrupt not being enabled if just one compatible device is configured (see [issue](https://github.com/zmkfirmware/zmk/issues/2201)).

Further in the current documentation, in the example the pins used for scanning the matrix are configured with `GPIO_PULL_DOWN`. This results of a current draw of 260mAh per configured pin during sleep. This is not necessary as the pull flags are set as needed during the scan.

### CODE REVIEW

File: docs/docs/config/kscan.md
Comment (caksoylar) [inline docs/docs/config/kscan.md:185]:
```suggestion
The [GPIO flags](https://docs.zephyrproject.org/3.5.0/hardware/peripherals/gpio.html#api-reference) for the elements in `gpios` should be `GPIO_ACTIVE_HIGH`, and interrupt pins set in `interrupt-gpios` should have the flags `(GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)`.
```

**Review by caksoylar — COMMENTED** [review]
Doc changes LGTM with the tweak below. Might be nice to split up the fix and docs changes to two commits.

**Comment by tobiasarndt** [comment]
> Might be nice to split up the fix and docs changes to two commits.

I split the commit. Is that what you meant, or should I make two different PRs?

**Comment by caksoylar** [comment]
Two commits instead of two PRs is fine by me, but ideally the commit messages'd fit within the [conventional commits guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary) that we follow (also you have three commits right now). This is easy to fix by maintainers when squashing and merging the PR but it isn't when multiple commits in the PR are preserved.

If you like I can fix this up for you before the PR is about to be completed, by pushing to your branch. But let's wait for feedback on the code fix first.

**Review by petejohanson — APPROVED** [review]
Thanks!

**Comment by caksoylar** [comment]
@petejohanson I rebased the changes to two separate commits, if it still looks good you can merge.

**Pull request merged**
